raft/raft.cpp

```diff
@@ -476,8 +477,56 @@ raft::handle_heartbeat_timeout(const boost::system::error_code& ec)
     std::lock_guard<std::mutex> lock(this->raft_lock);
 
     this->request_append_entries();
+    this->notify_leader_status();
 }
 
+void
+raft::notify_leader_status()
+{
+    if(!this->enable_audit)
+    {
+        return;
+    }
+
+    audit_message msg;
+    msg.mutable_leader_status()->set_term(this->current_term);
+    msg.mutable_leader_status()->set_leader(this->uuid);
+
+    bzn::message json;
+    json["bzn-api"] = "audit";
+    json["audit-data"] = boost::beast::detail::base64_encode(msg.SerializeAsString());
+
+    auto json_ptr = std::make_shared<bzn::message>(json);
+
+    for (const auto& peer : this->peers) {
+        auto ep = boost::asio::ip::tcp::endpoint{boost::asio::ip::address_v4::from_string(peer.host), peer.port};
+        this->node->send_message(ep, json_ptr);
+    }
+}
+
+void
+raft::notify_commit(size_t log_index, const std::string& operation)
+{
+    if(!this->enable_audit)
+    {
+        return;
+    }
+
+    audit_message msg;
+    msg.mutable_commit()->set_log_index(log_index);
+    msg.mutable_commit()->set_operation(operation);
+
+    bzn::message json;
+    json["bzn-api"] = "audit";
+    json["audit-data"] = boost::beast::detail::base64_encode(msg.SerializeAsString());
+
+    auto json_ptr = std::make_shared<bzn::message>(json);
+
+    for (const auto& peer : this->peers) {
+        auto ep = boost::asio::ip::tcp::endpoint{boost::asio::ip::address_v4::from_string(peer.host), peer.port};
+        this->node->send_message(ep, json_ptr);
```

Comment:
I wouldn't bother with the temporary and construct and endpoint in the send_message call.